src/main/java/com/fasterxml/jackson/databind/deser/impl/UnwrappedPropertyHandler.java

```diff
@@ -1,62 +1,112 @@
+    protected final Set<String> _unwrappedPropertyNames;
+
+    public UnwrappedPropertyHandler() {
+        _creatorProperties = new ArrayList<>();
+        _properties = new ArrayList<>();
+        _unwrappedPropertyNames = new HashSet<>();
+    }
 
-    public UnwrappedPropertyHandler()  {
-        _properties = new ArrayList<SettableBeanProperty>();
-   }
-    protected UnwrappedPropertyHandler(List<SettableBeanProperty> props)  {
+    protected UnwrappedPropertyHandler(List<SettableBeanProperty> creatorProps, List<SettableBeanProperty> props) {
+        _creatorProperties = creatorProps;
         _properties = props;
+        _unwrappedPropertyNames = Stream.concat(creatorProps.stream(), props.stream())
+                .map(SettableBeanProperty::getName)
+                .collect(Collectors.toSet());
+    }
+
+    public void addCreatorProperty(SettableBeanProperty property) {
+        _creatorProperties.add(property);
+        _unwrappedPropertyNames.add(property.getName());
     }
 
     public void addProperty(SettableBeanProperty property) {
         _properties.add(property);
+        _unwrappedPropertyNames.add(property.getName());
     }
 
-    public UnwrappedPropertyHandler renameAll(NameTransformer transformer)
-    {
-        ArrayList<SettableBeanProperty> newProps = new ArrayList<SettableBeanProperty>(_properties.size());
-        for (SettableBeanProperty prop : _properties) {
+    public UnwrappedPropertyHandler renameAll(NameTransformer transformer) {
+        return new UnwrappedPropertyHandler(
+                renameProperties(_creatorProperties, transformer),
+                renameProperties(_properties, transformer)
+        );
+    }
+
+    private List<SettableBeanProperty> renameProperties(
+            Collection<SettableBeanProperty> properties,
+            NameTransformer transformer
+    ) {
+        List<SettableBeanProperty> newProps = new ArrayList(properties.size());
+        for (SettableBeanProperty prop : properties) {
             String newName = transformer.transform(prop.getName());
             prop = prop.withSimpleName(newName);
             JsonDeserializer<?> deser = prop.getValueDeserializer();
             if (deser != null) {
                 @SuppressWarnings("unchecked")
-                JsonDeserializer<Object> newDeser = (JsonDeserializer<Object>)
-                    deser.unwrappingDeserializer(transformer);
+                JsonDeserializer<Object> newDeser = (JsonDeserializer<Object>) deser.unwrappingDeserializer(transformer);
                 if (newDeser != deser) {
                     prop = prop.withValueDeserializer(newDeser);
                 }
             }
             newProps.add(prop);
         }
-        return new UnwrappedPropertyHandler(newProps);
+        return newProps;
+    }
+
+    public boolean isUnwrapped(SettableBeanProperty property) {
+        return this._unwrappedPropertyNames.contains(property.getName());
+    }
```

Comment:
Would this be sufficient to check for the unwrapped properties? Not sure how different things affect the result of `getName()`

Reply:
In theory names should be `PropertyName` instances (matters with XML, mostly), but internally a few things are still just `String`s so I suspect this is fine.

Reply:
As per my comment higher up, I am not sure this method is actually needed: at least tests included seem to pass just fine without it being called (as well as all other existing tests).